src/tree/updater_quantile_hist.cc

```diff
@@ -871,27 +1200,45 @@ void QuantileHistMaker::Builder::EnumerateSplit(int d_step,
     iend = static_cast<int32_t>(cut_ptr[fid]) - 1;
   }
 
-  for (int32_t i = ibegin; i != iend; i += d_step) {
-    // start working
-    // try to find a split
-    e.Add(hist[i].GetGrad(), hist[i].GetHess());
-    if (e.sum_hess >= param_.min_child_weight) {
-      c.SetSubstract(snode.stats, e);
-      if (c.sum_hess >= param_.min_child_weight) {
-        bst_float loss_chg;
-        bst_float split_pt;
-        if (d_step > 0) {
-          // forward enumeration: split at right bound of each bin
-          loss_chg = static_cast<bst_float>(
-              spliteval_->ComputeSplitScore(nodeID, fid, e, c) -
-              snode.root_gain);
+  if (d_step == 1) {
+    for (int32_t i = ibegin; i < iend; i++) {
+      e.Add(hist[i].GetGrad(), hist[i].GetHess());
+      if (e.sum_hess >= param_.min_child_weight) {
+        c.SetSubstract(snode.stats, e);
+        if (c.sum_hess >= param_.min_child_weight) {
+          bst_float loss_chg;
+          bst_float split_pt;
+          {
+            loss_chg = static_cast<bst_float>(spliteval_->ComputeSplitScore(nodeID,
+              fid, e, c) - snode.root_gain);
+          }
+
           split_pt = cut_val[i];
           best.Update(loss_chg, fid, split_pt, d_step == -1, e, c);
```

Comment:
```suggestion
           best.Update(loss_chg, fid, split_pt, false, e, c);
```
since `d_step` is known to be 1 in this scope